pkg/network/dns_parser.go

```diff
@@ -195,7 +207,71 @@ func (*dnsParser) extractIPsInto(alias []byte, records []layers.DNSResourceRecor
 	}
 }
 
-func isWantedQueryType(checktype layers.DNSType) bool {
-	_, ok := recordedQueryTypes[checktype]
+func (p *dnsParser) isWantedQueryType(checktype layers.DNSType) bool {
+	_, ok := p.recordedQueryTypes[checktype]
 	return ok
 }
+
+func getRecordedQueryTypes(cfg *config.Config) map[layers.DNSType]struct{} {
+	if len(cfg.RecordedQueryTypes) <= 0 {
+		return defaultRecordedQueryTypes
+	}
+	queryTypes := make(map[layers.DNSType]struct{})
+	//
+	// check to see if we're recording more/different than the default
+	// query types
+	if len(cfg.RecordedQueryTypes) > 0 {
+
+		for _, t := range cfg.RecordedQueryTypes {
+			switch t {
```

Comment:
Rather than a giant switch statement, could you do something like this?
```
queryTypeStringMap := map[string]layer.DNSType{
	layers.DNSTypeA.String(): layers.DNSTypeA,
	layers.DNSTypeAAAA.String(): layers.DNSTypeAAAA,
	...
}
for _, t := range cfg.RecordedQueryTypes {
	  if qt, ok := queryTypeStringMap[t]; ok {
		  queryTypes[qt] = struct{}{}
	  } else {
		  log.Warnf("Unknown DNS type %v, skipping", t)
	  }
}
```